#!/usr/bin/env python
PACKAGE = "twistimu"

from dynamic_reconfigure.parameter_generator_catkin import *
from math import pi

gen = ParameterGenerator()

gen.add("pitch_min", double_t, 0, "", -2, -pi, pi)
gen.add("pitch_neutral", double_t, 0, "", 0, -pi, pi)
gen.add("pitch_max", double_t, 0, "", 2, -pi, pi)
gen.add("pitch_dead_zone", double_t, 0, "", 0.4, -pi, pi)
gen.add("roll_min", double_t, 0, "", -0.8, -pi, pi)
gen.add("roll_neutral", double_t, 0, "", 0.0, -pi, pi)
gen.add("roll_max", double_t, 0, "", 0.8, -pi, pi)
gen.add("roll_dead_zone", double_t, 0, "", 0.1, -pi, pi)
gen.add("speed_min", double_t, 0, "", -2, -100, 100)
gen.add("speed_neutral", double_t, 0, "", 0, -100, 100)
gen.add("speed_max", double_t, 0, "", 2, -100, 100)
gen.add("max_acceleration", double_t, 0, "maximum acceleration/deceleration allowed [m/s2]", 10, 0, 100)
gen.add("track", double_t, 0, "distance between parallel wheels [m]", 1, 0, 100)

exit(gen.generate(PACKAGE, "twistimu", "twistimu"))